pkgs/by-name/im/import-linter/package.nix

```
@@ -0,0 +1,36 @@
+{
+  lib,
+  python3Packages,
+  fetchPypi,
+}:
+
+python3Packages.buildPythonApplication rec {
+  pname = "import-linter";
+  version = "2.5.2";
+  pyproject = true;
+
+  src = fetchPypi {
+    inherit version;
+    pname = "import_linter";
+    hash = "sha256-2PLcZDKXXMNe3EzAv88bgR8FUAs3fODD9icp1o9Gxpg=";
+  };
+
+  nativeBuildInputs = with python3Packages; [
+    hatchling
+  ];
+
+  propagatedBuildInputs = with python3Packages; [
+    click
+    grimp
+    typing-extensions
+  ];
+
+  meta = with lib; {
+    description = "Enforces rules for the imports within and between Python packages";
+    homepage = "https://github.com/seddonym/import-linter/";
+    license = licenses.bsd2;
+    maintainers = with maintainers; [ sloschert ];
+    mainProgram = "lint-imports";
+  };
+
```

and the final nitpick
```suggestion
  meta = {
    description = "Enforces rules for the imports within and between Python packages";
    homepage = "https://github.com/seddonym/import-linter/";
    license = lib.licenses.bsd2;
    maintainers = with lib.maintainers; [ sloschert ];
    mainProgram = "lint-imports";
  };
```
`with` before attrsets (especially `with lib;`, _especially_ file-wide ones, _**especially**_ multiple ones) has been abused to death in the past and is now frowned upon[^1]. Just prepend `lib.` where necessary

[^1]: `with` before small scopes, like lists, is fine

---

Makes sense, thanks for the advise. I changed it.